proxy.go

```diff
@@ -73,7 +73,7 @@ func Join(local, remote net.Conn, log logging.Logger) {
 
 	transfer := func(side string, dst, src net.Conn) {
 		log.Debug("proxing %s -> %s", src.RemoteAddr(), dst.RemoteAddr())
-
+		defer dst.Close()
```

Comment:
Thanks for your help on this! I'm going to try to see if I can create a test case using `net/http` that reproduces this issue. 

What you're saying about the different types of io streams makes sense, but I'm confused why Line 82 (`err := src.Close();`) wouldn't be subject to the same kind of race condition? 

I'll try to let you know what I come up with sometime tonight.